README.md
# kylin-video

Video Player for Kylin OS.


参数配置文件：~/.config/kylin-video/kylin-video.ini
播放文件上次播放详细信息记录文件：~/.config/kylin-video/kylin_video_files.ini
播放驱动等信息记录文件：~/.config/kylin-video/player_info.ini

依赖并支持Mplayer和MPV两种视频播放引擎进行播放，可在设置中进行任意切换，默认使用
MPV。


Test URL:
http://rbv01.ku6.com/WrTcR6Yik9QIW4XIcxYvDw.mp4


debian/changelog
kylin-video (1.1.2-0ubuntu1) bionic; urgency=low

  *  Modifying the way to get the version information.

 -- lixiang <lixiang@kylinos.cn>  Tue, 06 Mar 2018 10:57:41 +0800

kylin-video (1.1.1kord) juniper; urgency=low

  * Edit Shortcut Key.

 -- lixiang <lixiang@kylinos.cn>  Wed, 13 Dec 2017 14:49:53 +0800

kylin-video (1.1.0kord) juniper; urgency=low

  * Add Stereo mode for audio.
  * Add frame rotation for video.
  * Show logo normal(LP: #4979).
  * Display the correct shortcut key of increase volume or decrease volume(LP: #4916).
  * When the playing video file is removed from the playlist, then stop playing it(LP: #4915).
  * Solved the mouse drag problem for playlist items(LP: #4907).
  * Add Shortcut for the button of setting audio delay(LP: #4917).

 -- lixiang <lixiang@kylinos.cn>  Tue, 26 Sep 2017 14:27:55 +0800

kylin-video (1.0.9kord) juniper; urgency=low

  * Support URL.
  * Optimize the aboutdialog.
  * Solved the problem that clicking on some buttons causes the interface to move sometimes.

 -- lixiang <lixiang@kylinos.cn>  Tue, 29 Aug 2017 16:48:53 +0800

kylin-video (1.0.8kord) juniper; urgency=low

  * Solve the background problem of pause button.
  * Optimize the scrollbar for playlist.
  * if the state is pause in unfullscreen, then show title bar and control bar.
  * if the state is not pause and playing in fullscreen, then show title bar and control bar.
  * Show normal window size when the icon on the taskbar is clicked, because of the bug of Qt5.

 -- lixiang <lixiang@kylinos.cn>  Wed, 09 Aug 2017 11:45:38 +0800

kylin-video (1.0.7kord) juniper; urgency=low

  * Set windown stay on top.
  * Add play order swither.
  * Add help page.
  * Add more shortcuts, such as full screen, playlist, about and exit.
  * Resolve the problem that text color settings is invalid of QComboBox.
  * Deal with the log records.

 -- lixiang <lixiang@kylinos.cn>  Fri, 28 Jul 2017 17:14:27 +0800

kylin-video (1.0.6kord) juniper; urgency=low

  * Add open screenshots action into systemtray.
  * Make the seek position work properly when using Mplayer.
  * Rewrite mainwindow with QStackedLayout.
  * Show detail info when the file does not exist(LP: #4372).
  * Resize TipWidget depending on the length of the tip text.
  * Show screenshot's information when it is saved successfully.
  * Add shortcut key notes.
  * Add Preview.

 -- lixiang <lixiang@kylinos.cn>  Mon, 24 Jul 2017 21:05:27 +0800

kylin-video (1.0.5kord) juniper; urgency=low

  * Add slot function for some actions.
  * Resolved the problem of selecting item is inaccurate when open files in different ways.
  * Modify progressbar.
  * Add error tip for the file has been removed from disk which double click play.
  * Add MessageDialog.
  * Modify QComboBox qss.
  * Remove Qt::ToolTip from TipWidget, make it be inside the mainwindow, but it's not transparent at this time.
  * Add tip when remove file from playlist.
  * Fixed some bugs.

 -- lixiang <lixiang@kylinos.cn>  Mon, 17 Jul 2017 15:57:20 +0800

kylin-video (1.0.4kord) juniper; urgency=low

  * Support MPV.
  * Support auto restart when the playback engine changed.

 -- lixiang <lixiang@kylinos.cn>  Wed, 05 Jul 2017 17:39:57 +0800

kylin-video (1.0.3kord) juniper; urgency=low

  * Update sub interfaces.
  * Update menu qss.

 -- lixiang <lixiang@kylinos.cn>  Tue, 04 Jul 2017 11:37:05 +0800

kylin-video (1.0.2kord) juniper; urgency=low

  * Update the main interface.

 -- lixiang <lixiang@kylinos.cn>  Tue, 20 Jun 2017 14:23:54 +0800

kylin-video (1.0.0kord) juniper; urgency=low

  * A great MPlayer front-end with Qt5.

 -- lixiang <lixiang@kylinos.cn>  Thu, 08 Jun 2017 16:57:19 +0800


src/version.cpp
/*  smplayer, GUI front-end for mplayer.
    Copyright (C) 2006-2015 Ricardo Villalba <rvm@users.sourceforge.net>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

#include "version.h"
#include <QApplication>

#define USE_SVN_VERSIONS 0

//#define VERSION "1.1.2"

QString Version::printable() {
    //return QString(VERSION);
    return QString(qApp->applicationVersion());
}

QString Version::stable() {
	//return QString(VERSION);
	return QString(qApp->applicationVersion());
}


src/main.cpp
/*  smplayer, GUI front-end for mplayer.
    Copyright (C) 2006-2015 Ricardo Villalba <rvm@users.sourceforge.net>

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

#include "myapplication.h"
#include "smplayer.h"

#include <QDir>
#include <QFile>
#include <QTextCodec>

int main( int argc, char ** argv ) 
{
    MyApplication a("kylin-video", argc, argv );
    a.setQuitOnLastWindowClosed(false);

    a.setOrganizationName("kylin");
    a.setApplicationName("kylin-video");
    a.setApplicationVersion("1.1.2");
	
#if QT_VERSION >= 0x040400
	// Enable icons in menus
	QCoreApplication::setAttribute(Qt::AA_DontShowIconsInMenus, false);
#endif

	QStringList args = a.arguments();

    QFile qss(":/qss/res/style.qss");
    qss.open(QFile::ReadOnly);
    qApp->setStyleSheet(qss.readAll());
    qss.close();

    QString arch = "";

#ifdef __x86_64__
        arch = "x86_64";
#elif __i386__
        arch = "i386";
#elif __aarch64__
        arch = "aarch64";
#endif

    SMPlayer * smplayer = new SMPlayer(arch);
    SMPlayer::ExitCode c = smplayer->processArgs(args);
	if (c != SMPlayer::NoExit) {
		return c;
	}
	smplayer->start();

	int r = a.exec();

	delete smplayer;

	return r;
}
